types/src/jwks/rsa.rs

```diff
@@ -37,8 +37,13 @@ impl RSA_JWK {
         }
     }
 
+    // TODO(zkid): Move this to aptos-crypto so other services can use this
     pub fn to_poseidon_scalar(&self) -> Result<ark_bn254::Fr> {
         let mut modulus = base64::decode_config(&self.n, URL_SAFE_NO_PAD)?;
+        // The circuit only supports RSA256
+        if modulus.len() != RSA_MODULUS_BYTES {
```

Comment:
Oops, I also added this check after fetching the JWK. I think it is best done there rather than in the hashing method?

Reply:
Personally I want to put it in the hashing method, we already have very circuit specific logic in the hashing method.